hbase-rest/src/test/java/org/apache/hadoop/hbase/rest/TestRESTServerSSL.java

```diff
@@ -0,0 +1,195 @@
+  private static final HBaseRESTTestingUtility REST_TEST_UTIL = new HBaseRESTTestingUtility();
+  private static Client sslClient;
+  private static File keyDir;
+  private Configuration conf;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    keyDir = initKeystoreDir();
+    KeyPair keyPair = KeyStoreTestUtil.generateKeyPair("RSA");
+    X509Certificate serverCertificate = KeyStoreTestUtil.generateCertificate(
+      "CN=localhost, O=server", keyPair, 30, "SHA1withRSA");
+
+    generateTrustStore("jks", serverCertificate);
+    generateTrustStore("jceks", serverCertificate);
+    generateTrustStore("pkcs12", serverCertificate);
+
+    generateKeyStore("jks", keyPair, serverCertificate);
+    generateKeyStore("jceks", keyPair, serverCertificate);
+    generateKeyStore("pkcs12", keyPair, serverCertificate);
+
+    TEST_UTIL.startMiniCluster();
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    TEST_UTIL.shutdownMiniCluster();
+  }
+
+  @Before
+  public void beforeEachTest() {
+    conf = new Configuration(TEST_UTIL.getConfiguration());
+    conf.set(Constants.REST_SSL_ENABLED, "true");
+    conf.set(Constants.REST_SSL_KEYSTORE_KEYPASSWORD, KEY_STORE_PASSWORD);
+    conf.set(Constants.REST_SSL_KEYSTORE_PASSWORD, KEY_STORE_PASSWORD);
+    conf.set(Constants.REST_SSL_TRUSTSTORE_PASSWORD, TRUST_STORE_PASSWORD);
+  }
+
+  @After
+  public void tearDownAfterTest() {
+    REST_TEST_UTIL.shutdownServletContainer();
+  }
+
+  @Test
+  public void testSslConnection() throws Exception {
+    startRESTServerWithDefaultKeystoreType();
+
+    Response response = sslClient.get("/version", Constants.MIMETYPE_TEXT);
+    assertEquals(200, response.getCode());
+  }
+
+  @Test(expected = org.apache.http.client.ClientProtocolException.class)
+  public void testNonSslClientDenied() throws Exception {
+    startRESTServerWithDefaultKeystoreType();
+
+    Cluster localCluster = new Cluster().add("localhost", REST_TEST_UTIL.getServletPort());
+    Client nonSslClient = new Client(localCluster, false);
+
+    nonSslClient.get("/version");
+  }
+
+  @Test
+  public void testSslConnectionUsingKeystoreFormatJKS() throws Exception {
+    startRESTServer("jks");
+
+    Response response = sslClient.get("/version", Constants.MIMETYPE_TEXT);
+    assertEquals(200, response.getCode());
+  }
+
+  @Test
+  public void testSslConnectionUsingKeystoreFormatJCEKS() throws Exception {
+    startRESTServer("jceks");
+
+    Response response = sslClient.get("/version", Constants.MIMETYPE_TEXT);
+    assertEquals(200, response.getCode());
+  }
+
+  @Test
+  public void testSslConnectionUsingKeystoreFormatPKCS12() throws Exception {
+    startRESTServer("pkcs12");
+
+    Response response = sslClient.get("/version", Constants.MIMETYPE_TEXT);
+    assertEquals(200, response.getCode());
+  }
+
+
+
+  private static File initKeystoreDir() {
+    final File target = new File(System.getProperty("user.dir"), "target");
```

Comment:
Please create a temporary directory and remove it in `@AfterClass`.

Reply:
this was a copy-paste leftover, thanks for noticing! I'll remove these two lines.
(I'm using the `TEST_UTIL.getDataTestDir()` which get deleted when the minicluster is stopped in the `@AfterClass` method)